src/fetch-data.js

```diff
@@ -1,16 +1,30 @@
-import React, { Component } from 'react'
+/* eslint-disable react/prop-types */
+// I  have prop types here but eslint doesn't seem to pick it up
+import React, { Component, PropTypes } from 'react'
 import AsyncProps from 'async-props'
+import { has } from 'lodash'
+import MissingView from './shared/missing-view'
 import fetchRouteData from './shared/fetch-route-data'
 
-export default  function(TopLevelComponent) {
-  return class DataHOC extends Component {
+export default function fetchData(TopLevelComponent) {
+  class FetchDataHOC extends Component {
     static loadProps({ params, loadContext }, cb) {
       // Don't need any IDs or anything we just use the name of the component `Post, Category, FrontPage` etc
-      return fetchRouteData(TopLevelComponent.constructor.name, { params, loadContext, cb })
+      return fetchRouteData(TopLevelComponent.name, { params, loadContext, cb })
     }
 
     render() {
+      const Error = MissingView
```

Comment:
Ah yes this is a good point, and something I forgot to note down.

The custom `Error` or (`CustomError` perhaps!) in this single loader is problematic, how do we get custom error components in.

`function fetchData(Component, params => 'posts?filter=${params.tag}', ErrorComponent)`